### PR TITLE
Add support for SurfaceTexture based external textures on Android under Impeller/GLES.

### PR DESCRIPTION
**Pull request by johnmccutchan**
- Refactor AndroidExternalTextureGL into SurfaceTextureExternalTexture base class with two specialized implementations:
- Add SurfaceTextureExternalTextureGL for legacy GL rendering.
- Add SurfaceTextureExternalTextureImpellerGL for Impeller GLES rendering.

NOTE: There is still no support for SurfaceTextures under Vulkan rendering.

### CODE REVIEW

File: shell/platform/android/surface_texture_external_texture.h [hunk 1]
```diff
@@ -1,9 +1,10 @@
+
```
Comment (jonahwilliams) [inline shell/platform/android/surface_texture_external_texture.h:1]:
```suggestion
```

File: shell/platform/android/surface_texture_external_texture_gl.cc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+void SurfaceTextureExternalTextureImpellerGL::ProcessFrame(
+    PaintContext& context,
+    const SkRect& bounds) {
+  if (state_ == AttachmentState::kUninitialized) {
+    // Generate the texture handle.
+    impeller::TextureDescriptor desc;
+    desc.type = impeller::TextureType::kTextureExternalOES;
+    desc.storage_mode = impeller::StorageMode::kDevicePrivate;
+    desc.format = impeller::PixelFormat::kR8G8B8A8UNormInt;
```
Comment (jonahwilliams) [inline shell/platform/android/surface_texture_external_texture_gl.cc:86]: Same comment as last PR: we need to look this up at runtime right?

File: shell/platform/android/platform_view_android.cc [hunk 3]
```diff
@@ -301,10 +301,20 @@ void PlatformViewAndroid::RegisterExternalTexture(
     int64_t texture_id,
     const fml::jni::ScopedJavaGlobalRef<jobject>& surface_texture) {
   if (android_context_->RenderingApi() == AndroidRenderingAPI::kOpenGLES) {
-    RegisterTexture(std::make_shared<AndroidExternalTextureGL>(
-        texture_id, surface_texture, jni_facade_));
+    if (android_context_->GetImpellerContext() != nullptr) {
```
Comment (jonahwilliams) [inline shell/platform/android/platform_view_android.cc:304]: I think we'd probably just write it like
```suggestion
    if (android_context_->GetImpellerContext()) {
```

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: shell/platform/android/surface_texture_external_texture_gl.cc (hunk 2 above)
Comment (jonahwilliams) [reply shell/platform/android/surface_texture_external_texture_gl.cc:86]: Interesting! right, I guess its GL. I think we also do something similar in the GLES Impeller backend.

File: shell/platform/android/surface_texture_external_texture.h (hunk 1 above)
Comment (johnmccutchan) [reply shell/platform/android/surface_texture_external_texture.h:1]: Done.

File: shell/platform/android/platform_view_android.cc (hunk 3 above)
Comment (johnmccutchan) [reply shell/platform/android/platform_view_android.cc:304]: Done.

**Pull request merged**